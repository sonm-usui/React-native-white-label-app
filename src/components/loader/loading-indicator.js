// @flow
import React, { PureComponent } from 'react'
import { ActivityIndicator } from 'react-native'
import type { LoadingIndicatorProps } from './type-loader'
import { DARK } from './type-loader'
import { color } from '../../common/styles/constant'

export class LoadingIndicator extends PureComponent<
  LoadingIndicatorProps,
  void
> {
  static defaultProps = {
    type: DARK,
  }

  render() {
    const { type } = this.props
    const tintColor = type === DARK ? color.actions.sixth : color.actions.none

    return <ActivityIndicator color={tintColor} size="large" hidesWhenStopped />
  }
}
