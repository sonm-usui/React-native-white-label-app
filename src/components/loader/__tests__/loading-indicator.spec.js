// @flow
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { LoadingIndicator } from '../loading-indicator'

describe('<LoadingIndicator />', () => {
  beforeEach(() => {
    jest.useFakeTimers()
  })

  afterEach(() => {
    jest.runOnlyPendingTimers()
  })

  it('should render properly and match the snapshot', () => {
    const component = renderer.create(<LoadingIndicator size={30} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
