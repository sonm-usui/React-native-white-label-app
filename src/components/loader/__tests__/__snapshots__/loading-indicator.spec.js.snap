// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> should render properly and match the snapshot 1`] = `
<ActivityIndicator
  color="#777777"
  hidesWhenStopped={true}
  size="large"
/>
`;
